src/librustdoc/html/static/main.js

```diff
@@ -898,72 +855,34 @@ function hideThemeButtonState() {
         return "\u2212"; // "\u2212" is "−" minus sign
     }
 
-    function onEveryMatchingChild(elem, className, func) {
-        if (elem && className && func) {
-            var length = elem.childNodes.length;
-            var nodes = elem.childNodes;
-            for (var i = 0; i < length; ++i) {
-                if (hasClass(nodes[i], className)) {
-                    func(nodes[i]);
-                } else {
-                    onEveryMatchingChild(nodes[i], className, func);
-                }
-            }
-        }
-    }
-
-    function toggleAllDocs(fromAutoCollapse) {
+    function toggleAllDocs() {
         var innerToggle = document.getElementById(toggleAllDocsId);
         if (!innerToggle) {
             return;
         }
+        var sectionIsCollapsed = false;
         if (hasClass(innerToggle, "will-expand")) {
             removeClass(innerToggle, "will-expand");
-            onEachLazy(document.getElementsByTagName("details"), function(e) {
-                e.open = true;
-            });
-            onEveryMatchingChild(innerToggle, "inner", function(e) {
-                e.innerHTML = labelForToggleButton(false);
+            onEachLazy(document.getElementsByClassName("rustdoc-toggle"), function(e) {
+                if (hasClass(e, "type-contents-toggle") === false) {
```

Comment:
I think you can just use `!hasClass(e, "type-contents-toggle")`. But also, why is `type-contents-toggle` excluded from this? And why is `implementors-toggle` excluded below?

Reply:
We are less picky on what we "expand" compared to what we "collapse". Also, the type declaration is always excluded from the global toggle. I mostly aimed at keeping the same behaviour but we can discuss about changing it after this PR is merged.

Reply:
Updated to `!hasClass(e, "type-contents-toggle")`.